test/test.googleauth.ts

```diff
@@ -15,20 +15,36 @@
  */
 
 import * as assert from 'assert';
+import {AxiosPromise, AxiosRequestConfig, AxiosResponse} from 'axios';
 import * as fs from 'fs';
+import * as http from 'http';
 import * as nock from 'nock';
 import * as path from 'path';
-import * as request from 'request';
+import * as stream from 'stream';
 
-import {Compute} from '../src/auth/computeclient';
 import {GoogleAuth} from '../src/auth/googleauth';
 import {JWT} from '../src/auth/jwtclient';
-import {OAuth2Client} from '../src/auth/oauth2client';
 import {UserRefreshClient} from '../src/auth/refreshclient';
 import {BodyResponseCallback, DefaultTransporter} from '../src/transporters';
 
 nock.disableNetConnect();
 
+afterEach(() => {
+  nock.cleanAll();
+});
+
+function createIsGCENock(isGCE = true) {
+  nock('http://metadata.google.internal').get('/').reply(200, null, {
+    'metadata-flavor': 'Google'
+  });
+}
+
+function createGetProjectIdNock(projectId: string) {
+  nock('http://169.254.169.254')
```

Comment:
I opened #190 to track.  There was some interesting context in the code on why it's using the IP:

` Uses 169.254.169.254 for the metadata server to avoid request latency from DNS lookup. See https://cloud.google.com/compute/docs/metadata#metadataserver for information about this IP address. (This IP is also used for Amazon EC2 instances, so the metadata flavor is crucial.)`

Reply:
I know about that one. We are not consistent internally though.

I am doubtful that the optimization was worthwhile though; and I suspect it might be environment specific (k8s, is my guess). A better fix would have been to try resolving `metadata.google.internal.` with the trailing dot. Regardless, let's address it in #190.